### PR TITLE
[Core] Fix Transform::log() and SpatialMotionVector::exp()

### LINKED ISSUE #561: [Core] iDynTree::Transform::log() and iDynTree::SpatialMotionVector::exp() computations seems to be wrong?
Here, the log of SE(3) says the linear part is unchanged 
https://github.com/robotology/idyntree/blob/7733e56f858cfb319261f118871d37b8cbc5ca0f/src/core/src/Transform.cpp#L534

which does not seem to be coherent with the definition of SE(3) logarithm described in [Appendix section B  of the paper Associating Uncertainty With Three-Dimensional Poses for Use in Estimation Problems](http://ncfrn.mcgill.ca/members/pubs/barfoot_tro14.pdf).

Looks like we are missing a multiplication of the linear part with a left/right Jacobian inverse of $SO(3)$.

where, the left Jacobian of $SO(3)$ is defined as,

$$ J_{l_{SO(3)}} = \sum_{n = 0}^{\infty} \frac{1}{(n+1)!} [\omega_\times]^n  = (I_3 + \frac{1 - \text{cos}(||\omega||)}{||\omega||^{2}} [\omega _{\times}] + \frac{||\omega|| - \text{sin}(||\omega||)}{||\omega||^{3}} [\omega _{\times}]^{2})$$ where $\omega$ is the `AngularMotionVector3`.

When simplified further, $$ J_{l_{SO(3)}} = \frac{\text{sin}(||\omega||)}{||\omega||}I_3 + \frac{1 - \text{cos}(||\omega||)}{||\omega||} [\phi _{\times}] + \bigg(1 - \frac{\text{sin}(||\omega||)}{||\omega||}\bigg) \phi\phi^T$$ where $\phi = \frac{\omega}{||\omega||}$

The inverse left Jacobian is given by,
$$ J^{-1} _{l _{SO(3)}} = \frac{||\omega||}{2} \text{cot} \bigg(\frac{||\omega||}{2}\bigg) I _3 + \bigg( 1 - \frac{||\omega||}{2} \text{cot} \bigg(\frac{||\omega||}{2}\bigg) \bigg) \phi \phi^T - \frac{||\omega||}{2} [\phi _{\times}]$$

cc @traversaro

### PR DESCRIPTION
**Pull request by prashanthr05**
- Adds left Jacobian and left Jacobian inverse of SO(3) to Rotation
- Fixes Transform::log()
- Fixes SpatialMotionVector::exp()

### CODE REVIEW

**Comment by prashanthr05** [comment]
@traversaro it'd be great if you could review this PR.

File: src/core/tests/TransformFromMatrix4x4UnitTest.cpp [hunk 1]
```diff
@@ -36,8 +37,26 @@ void checkFromMatrix4x4()
...
+    Position pos = Position(pos_raw);
+    Transform t_posrot(rot, pos);
+
+    SpatialMotionVector v;
+    v.zero();
+    v.setAngularVec3(rot.log());
+
+    Transform t_v = v.exp();
+
+    ASSERT_EQUAL_TRANSFORM_TOL(t_posrot, t_v, 0.01);
```
Comment (traversaro) [inline src/core/tests/TransformFromMatrix4x4UnitTest.cpp]: This test is not testing the linear part?

Comment (prashanthr05) [reply src/core/tests/TransformFromMatrix4x4UnitTest.cpp]: Computationally, it is. Quantitatively, no. Since the linear part is set to all zeros.

Comment (traversaro) [reply src/core/tests/TransformFromMatrix4x4UnitTest.cpp]: I see, we are not checking if the function is correct, but at least we are checking stuff like that memory is correct with Valgrind, etc etc.

**Comment by traversaro** [comment]
I am bit puzzled about that. In theory changing the behavior of a public function would call for creating a new function and  deprecate the old one, but in practice the behavior of those function was not documented, so I think it is fair to consider the "wrong" behavior for the linear part as a bug.

**Comment by prashanthr05** [comment]
> I am bit puzzled about that. In theory changing the behavior of a public function would call for creating a new function and deprecate the old one, but in practice the behavior of those function was not documented, so I think it is fair to consider the "wrong" behavior for the linear part as a bug.

I do not fully comprehend this statement. You mean since some macro related to make_deprecated was not called since we made changes directly on the public API of Transform or SpatialMotionVector?

**Comment by traversaro** [comment]
> I do not fully comprehend this statement. You mean since some macro related to make_deprecated was not called since we made changes directly on the public API of Transform or SpatialMotionVector?

I will explain to you in person. In a nutshell, in general you cannot change the behaviour of a public function, i.e. if you have a function `int sum(int, int)` that sums two integers, you cannot suddenly change its behaviour to something else, unless the previous behaviour was not consistent with what documented/had a bug.

**Comment by prashanthr05** [comment]
> you cannot suddenly change its behaviour to something else, unless the previous behaviour was not consistent with what documented/had a bug.

aAh, okay. I understand and I agree.

**Pull request merged**